lib/manager/composer/utils.ts

```diff
@@ -1,17 +1,57 @@
+import { getPkgReleases } from '../../datasource';
 import { logger } from '../../logger';
 import { api, id as composerVersioningId } from '../../versioning/composer';
 import type { ComposerConfig, ComposerLock } from './types';
 
 export { composerVersioningId };
 
-export function getConstraint(constraints: Record<string, string>): string {
+export async function getComposerConstraint(
+  constraints: Record<string, string>
+): Promise<string> {
   const { composer } = constraints;
 
-  if (composer) {
-    logger.debug('Using composer constraint from config');
+  if (api.isSingleVersion(composer)) {
+    logger.debug(
+      { version: composer },
+      'Using composer constraint from config'
+    );
     return composer;
   }
 
+  const release = await getPkgReleases({
+    depName: 'composer/composer',
+    datasource: 'github-releases',
+    versioning: composerVersioningId,
+  });
+
+  if (!release?.releases?.length) {
+    throw new Error('No composer releases found.');
+  }
+  let versions = release.releases.map((r) => r.version);
+
+  if (composer) {
+    versions = versions.filter(
+      (v) => api.isValid(v) && api.matches(v, composer)
+    );
+  }
+
+  if (!versions.length) {
+    throw new Error('No compatible composer releases found.');
```

Comment:
Any way to fall back to latest? Or best we abort?

Reply:
Abort, as this must be a temporary error.